src/Popover/index.js

```diff
@@ -53,12 +57,11 @@ const Popover = ({
 
   const togglePopover = (event) => {
     event.stopPropagation();
-    setOpen((open) => {
-      if (!open) {
-        onUserEnable();
-      }
-      return !open;
-    });
+    if (isControlled) {
+      isOpen ? onUserDismiss() : onUserEnable();
+    } else {
+      setOpen((open) => !open);
+    }
```

Comment:
Fixes a bug where this one codepath wasn't called the dismissal callback as expected